docs/design.md

```diff
@@ -0,0 +1,872 @@
+        configMap:
+          name: stash-sample-data
+```
+
+
+
+### Default Backup of a PVC
+
+Sample `DefaultBackupConfiguration` crd for stand-alone pvc backup:
+
+```yaml
+apiVersion: stash.appscode.com/v1alpha2
+kind: DefaultBackupConfiguration
+metadata:
+  name: default-volume-backup-configuration
+spec:
+  backend:
+    gcs:
+      bucket: stash-backup-repo
+      prefix: ${metadata.namespace}/${metadata.name} # this prefix template will be used to initialize repository in different directory in backend.
+    storageSecretName: gcs-secret
+  schedule: '@every 1h'
+  stashTemplate: volumeBackup
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Sample PVC with annotation for default backup:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: demo-pvc
+  namespace: demo
+  # if stash find bellow annotations, it will take backup of it.
+  annotations:
+    stash.appscode.com/backup: true
+    stash.appscode.com/stashTemplate: "volumeBackup"
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+
+
+### Default Backup of Database
+
+Sample `DefaultBackupConfiguration` crd for database backup:
+
+```yaml
+apiVersion: stash.appscode.com/v1alpha2
+kind: DefaultBackupConfiguration
+metadata:
+  name: database-default-backup-configuration
+spec:
+  backend:
+    gcs:
+      bucket: stash-backup-repo
+      prefix: ${metadata.namespace}/${metadata.name} # this prefix template will be used to initialize repository in different directory in backend.
+    storageSecretName: gcs-secret
+  schedule: '@every 1h'
+  stashTemplate: pgBackup
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Sample `AppBinding` crd with annotations for default backup:
+
+```yaml
+apiVersion: appcatalog.appscode.com/v1alpha1
+kind: AppBinding
+metadata:
+  name: quick-postgres
+  namespace: demo
+  labels:
+    kubedb.com/kind: Postgres
+    kubedb.com/name: quick-postgres
+    # if stash find bellow annotations, it will take backup of it.
+    annotations:
+      stash.appscode.com/backup: true
+      stash.appscode.com/stashTemplate: "pgBackup"
+spec:
+  clientConfig:
+    insecureSkipTLSVerify: true
+    service:
+      name: quick-postgres
+      port: 5432
+      scheme: "http"
+  secret:
+    name: quick-postgres-auth
+  type: kubedb.com/postgres
+```
+
+
+
+**How it will work?**
+
+- Stash will watch the workloads, volume and `AppBinding` crds. When Stash will find an workload/volume/AppBinding crd with these annotations, it will create a `Repository` crd and a `BackupConfiguration` crd using the information from respective `StashTemplate`.
+- Then, Stash will take normal backup as discussed earlier.
+
+
+
+## Auto Recovery
+
+User will be also able to configure an automatic recovery for a particular workload. Each time the workload restart, at first it will perform recovery then original workload's container will start.
```

Comment:
This should be an annotation on the workload not a attribute of `RestoreConfiguration`.